driver/driver.feature

```diff
@@ -0,0 +1,1265 @@
+#    When transaction closes
+#    Then connection has database: created-after-schema
+#    Then connection does not have database: created-after-write
+#    Then connection does not have database: created-after-read
+#
+#    When connection open write transaction for database: typedb
+#    Then connection has database: created-after-schema
+#    Then connection does not have database: created-after-write
+#    Then connection does not have database: created-after-read
+#    When transaction closes
+#    Then connection has database: created-after-schema
+#    Then connection does not have database: created-after-write
+#    Then connection does not have database: created-after-read
+#
+#    When connection open write transaction for database: typedb
+#    When schedule database creation on transaction close: created-after-write
+#    Then connection has database: created-after-schema
+#    Then connection does not have database: created-after-write
+#    Then connection does not have database: created-after-read
+#    When transaction closes
+#    Then connection has database: created-after-schema
+#    Then connection has database: created-after-write
+#    Then connection does not have database: created-after-read
+#
+#    When connection open read transaction for database: typedb
+#    When schedule database creation on transaction close: created-after-read
+#    Then connection has database: created-after-schema
+#    Then connection has database: created-after-write
+#    Then connection does not have database: created-after-read
+#    When transaction closes
+#    Then connection has database: created-after-schema
+#    Then connection has database: created-after-write
+#    Then connection has database: created-after-read
+
+
+  Scenario: Driver processes transaction commit errors correctly
+    Given connection open schema transaction for database: typedb
+    When typeql schema query
+      """
+      define attribute name;
+      """
+    Then transaction commits; fails
+
+
+  # TODO: check errors on transaction commits with callbacks (on_close) set!
+
+  ###########
+  # QUERIES #
+  ###########
+
+  Scenario: Driver processes ok query answers correctly
+    Given connection open schema transaction for database: typedb
+    When get answers of typeql schema query
+      """
+      define entity person;
+      """
+    Then answer type is: ok
+    Then answer type is not: concept rows
+    Then answer type is not: concept trees
+    Then result is a successful ok
+    Then transaction commits
+
+
+    # TODO: Test optionals when introduced
+  Scenario: Driver processes concept row query answers correctly
+    Given connection open schema transaction for database: typedb
+    Given typeql schema query
+      """
+      define entity person owns name @key; attribute name, value string;
+      """
+    When get answers of typeql read query
+      """
+      match entity $p;
+      """
+    Then answer type is: concept rows
```

Comment:
Initially, I liked "is concept rows" and "is not concept rows". However, it didn't want to work with the python's bdds parser when I expected any value type (for created structs) and had "is/is not" on the right without a specific separator. So I ended up using "is:" and "is not:" everywhere.